tests/system/src/test/java/test/robot/javafx/stage/FocusParentWindowOnChildCloseTest.java

```diff
@@ -0,0 +1,159 @@
+        waitForLatch(startupLatch, 10, "FX runtime failed to start.");
+    }
+
+    @AfterClass
+    public static void exit() {
+        Platform.runLater(() -> {
+            stage.hide();
+            stage2.hide();
+        });
+        Platform.exit();
+    }
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                    (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    private static void fireOkInAlert() throws Exception {
+        alertCloseLatch = new CountDownLatch(1);
+        Platform.runLater(() -> {
+                    Button okButton = (Button) alert.getDialogPane().lookupButton(ButtonType.OK);
+                    System.out.println(okButton);
+                    okButton.fire();
+                });
+        Thread.sleep(400);
+        waitForLatch(alertCloseLatch, 10, "Failed to close alert.");
+    }
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            robot = new Robot();
+            stage = primaryStage;
+
+            createWindow1(stage);
+            createWindow2(stage);
+        }
+
+        private void createWindow1(Stage stage) {
+            stage.setTitle("Stage - 1");
+            button = new Button();
+            button.setText("Show Alert");
+            button.setOnAction(event -> showAlert(stage));
+
+            StackPane root = new StackPane();
+            root.getChildren().add(button);
+            scene = new Scene(root, 300, 250);
+            stage.setScene(scene);
+            stage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> Platform.runLater(startupLatch::countDown));
+            stage.show();
+        }
+
+        private void showAlert(Stage primaryStage) {
+            alert = new Alert(Alert.AlertType.CONFIRMATION);
+            alert.initOwner(primaryStage);
+            alert.setOnShown(event -> Platform.runLater(alertShownLatch::countDown));
+            alert.setOnHidden(event -> Platform.runLater(alertCloseLatch::countDown));
+            alert.show();
+        }
+
+        private void createWindow2(Stage stage1) {
+            stage2 = new Stage();
+            stage2.setX(stage1.getX() + 300);
+            stage2.setY(stage1.getY());
+
+            stage2.setTitle("Stage - 2");
+            StackPane root = new StackPane();
+            stage2.setScene(new Scene(root, 300, 250));
+            stage2.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> Platform.runLater(startupLatch::countDown));
+            stage2.show();
+        }
+    }
+
+    public static void waitForLatch(CountDownLatch latch, int seconds, String msg) throws Exception {
+        org.junit.Assert.assertTrue("Timeout: " + msg, latch.await(seconds, TimeUnit.SECONDS));
```

Comment:
Minor nit: you can just use `Assert` rather than the fully-qualified class, since it is imported.

Reply:
done!